yaks/lib/yaks/behaviour/optional_includes.rb

```diff
@@ -0,0 +1,29 @@
+require "rack/utils"
+
+module Yaks
+  module Behaviour
+    module OptionalIncludes
+      RACK_KEY = "yaks.optional_includes".freeze
+
+      def associations
+        super.select do |association|
+          association.if != Undefined || include_association?(association)
```

Comment:
it would at this stage, `map_associations` handles evaluating the if and taking the association out